src/storage/preferences.rs

```diff
@@ -67,11 +56,18 @@ impl DBPreferences {
 
 impl Default for DBPreferences {
     fn default() -> Self {
+        let version_str = env!("CARGO_PKG_VERSION");
+        let db_version = match LogVersion::try_from(version_str) {
+            Ok(version) => version,
+            Err(_err) => LogVersion::new(0, 0, 0),
```

Comment:
Should `db_version` be a preference at all? I mean, if the code can only handle `2.0.0` log format, what's the meaning of the user setting the db_version to anything else (say `3.0.0` or `1.0.0`)?

I think `db_version` might as well just be a fixed struct, directly controller by `CARGO_PKG_VERSION` and cannot be changed or specified manually by user.

Reply:
done